go/ephemeral/lib.go

```diff
@@ -393,6 +405,45 @@ func (e *EKLib) PrepareNewUserEK(ctx context.Context, merkleRoot libkb.MerkleRoo
 	return prepareNewUserEK(ctx, e.G(), merkleRoot, signingKey)
 }
 
+func (e *EKLib) BoxLatestTeamEK(ctx context.Context, teamID keybase1.TeamID, recipients []keybase1.UID) (teamEKBoxes *[]keybase1.TeamEkBoxMetadata, err error) {
+	defer e.G().CTrace(ctx, "BoxLatestTeamEK", func() error { return err })()
+
+	// TODO remove this when we want to release in the wild.
+	if !e.ShouldRun(ctx) {
+		return nil, nil
+	}
+
+	// If we need a new teamEK let's just create it when needed, the new
+	// members will be part of the team and will have access to it via the
+	// normal mechanisms.
```

Comment:
We'll probably want to add a check on the server at some point (once support is widespread) that new team members are always boxed for the latest key. That probably means we'd either want to send a box along for the ride, or rotate the whole teamEK as part of the transaction. Will this short-circuit stymie that check?

Reply:
I think that's true for the userEK/deviceEK case but for teams since we only generate on demand it may not hold. what do you think?

Reply:
Hmm good point. We can't enforce reboxing, because the admin here might've just de-closeted. But we probably also don't want to force that admin to roll a whole new key just to add a member.

By the way, in that case, where the team key is current but the admin here doesn't have a copy, is that going to fail the new member add?

Reply:
that's the point of this check though. it will return `true` since they can't access the latest key but then short circuit and not try to rebox